src/NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvalues.cpp

```diff
@@ -0,0 +1,99 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/LinearAlgebra/FindGeneralizedEigenvalues.hpp"
+
+#include <cstddef>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Matrix.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "ErrorHandling/Error.hpp"
+#include "Utilities/Gsl.hpp"
+
+// LAPACK routine to do the generalized eigenvalue problem
+extern "C" {
+extern void dggev_(char*, char*, int*, double*, int*, double*, int*, double*,
+                   double*, double*, double*, int*, double*, int*, double*,
+                   int*, int*);
+}
+
+void find_generalized_eigenvalues(
+    gsl::not_null<DataVector*> eigenvalues_real_part,
+    gsl::not_null<DataVector*> eigenvalues_imaginary_part,
+    gsl::not_null<Matrix*> eigenvectors, const Matrix& A,
+    const Matrix& B, const bool force_lapack_failure) noexcept {
+  // Sanity checks on the sizes of the vectors and matrices
+  const size_t number_of_rows = A.rows();
+  ASSERT(number_of_rows == A.columns(), "Matrix A should be square");
+  ASSERT(number_of_rows == B.rows() and number_of_rows == B.columns(),
+         "Matrix A and matrix B should be the same size");
+  ASSERT(number_of_rows == eigenvectors->rows() and
+             number_of_rows == eigenvectors->columns(),
+         "Matrix A and matrix eigenvectors should have the same size");
+  ASSERT(number_of_rows == eigenvalues_real_part->size() and
+             number_of_rows == eigenvalues_imaginary_part->size(),
+         "eigenvalues DataVector sizes should equal number of columns "
+         "in Matrix A");
+
+  // Set up parameters for the lapack call
+  // Lapack uses chars to decide whether to compute the left eigenvectors,
+  // the right eigenvectors, both, or neither. 'N' means do not compute,
+  // 'V' means do compute.
+  auto compute_left_eigenvectors = 'N';
+  auto compute_right_eigenvectors = 'V';
+
+  // Lapack overwrites the input matrices, so copy the inputs into
+  // non-const matrices
+  auto A_for_lapack = A;
+  auto B_for_lapack = B;
+
+  // Lapack expects the sizes to be ints, not size_t.
+  // NOTE: not const because lapack function dggev_() arguments
+  // are not const.
+  auto matrix_and_vector_size = static_cast<int>(number_of_rows);
+
+  // Lapack splits the eigenvalues into unnormalized real and imaginary
+  // parts, which it calls alphar and alphai, and a normalization,
+  // which it calls beta. The real and imaginary parts of the eigenvalues are
+  // found by dividing the unnormalized results by the normalization.
+  DataVector unnormalized_eigenvalue_real_part(number_of_rows, 0.0);
+  DataVector unnormalized_eigenvalue_imaginary_part(number_of_rows, 0.0);
```

Comment:
get rid of the `unnormalized` ones, just pass in the `ev_real_part`, imaginary part, then do a `/= eigenvalue_normalization`